### PR TITLE
✨CGS Update part 2 (Remastered)

### PR DESCRIPTION
**Pull request by Szymekk44**
This time nice and clean!
Added 
- GetImage (SVGAII & VBE)
- DrawCroppedImage (You can now draw wallpaper cropped by your taskbar height!)
- DrawPoint (int)
- GetRawPoint
- DrawRectangle has been improved
- More draw array functions - for int[] (just copying to video buffer!)
- MANY VBE Bugfixes
- GetImage tests
- Finished preventOffBoundPixels (now its working for left side and right side - finished part of CGS Update part 1)

Note: I didn't add XML to the override voids, as the others don't have them either. XML comments are always above virtual/abstract voids. Writing this because of my last pr hihi

### CODE REVIEW

File: source/Cosmos.System2/Graphics/FullScreenCanvas.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ private enum VideoDriver
             VGADriver
         }
 
-        static Canvas videoDriver = null;
+        public static Canvas videoDriver = null;
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/FullScreenCanvas.cs:38]: Why do we need this to be public?

File: source/Cosmos.System2/Graphics/SVGAIICanvas.cs [hunk 2]
```diff
@@ -17,8 +17,8 @@ public class SVGAIICanvas : Canvas
         internal Debugger debugger = new("SVGAIIScreen");
         static readonly Mode defaultMode = new(1024, 768, ColorDepth.ColorDepth32);
 
-        private Mode mode;
-        private readonly VMWareSVGAII driver;
+        public Mode mode;
+        public readonly VMWareSVGAII driver;
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/SVGAIICanvas.cs:21]: Same question here

File: source/Cosmos.System2/Graphics/VBECanvas.cs [hunk 3]
```diff
@@ -14,8 +14,8 @@ namespace Cosmos.System.Graphics
     public class VBECanvas : Canvas
     {
         static readonly Mode defaultMode = new(1024, 768, ColorDepth.ColorDepth32);
-        readonly VBEDriver driver;
-        Mode mode;
+        public readonly VBEDriver driver;
+        public Mode mode;
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/VBECanvas.cs:18]: Same

**Review by quajak — CHANGES_REQUESTED** [review]
Thank you for the work. The code is very clean! Just this one repeated question about why we are making certain variables public

File: source/Cosmos.System2/Graphics/FullScreenCanvas.cs (hunk 1 above)
Comment (Szymekk44) [reply source/Cosmos.System2/Graphics/FullScreenCanvas.cs:38]: so you can directly access framebuffer and many driver stuff
- CopyVRAM
- GetVRAM
- SetVRAM
- ClearVRAM
- Fill

Also i think that people should have direct access to grapic driver, only positives

**Review by quajak — APPROVED** [review]

**Pull request merged**